### PR TITLE
feat: upload cosign public key and notation certificates to cloud

### PR DESCRIPTION
**Pull request by Andreea-Lupu**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/meta/signatures/signatures.go [hunk 1]
```diff
@@ -21,18 +29,106 @@ const (
...
+	}
+
+	return &SigStore{
+		CosignStorage:   publicKeyStorage,
+		NotationStorage: certStorage,
+	}, nil
+}
+
+func GetSecretsManagerClient(region, endpoint string) (*secretsmanager.Client, error) {
+	customResolver := aws.EndpointResolverWithOptionsFunc(
```
Comment (peusebiu) [inline pkg/meta/signatures/signatures.go:79]: we already have this logic duplicated for dynamodb client, you can split both GetDynamoClient and GetSecretsManagerClient in two parts,
func GetAwsConfig(region, endpoint) aws.Config {}
+
func GetDynamoClient(awsConfig) {}
func GetSecretsManagerClient(awsConfig) {}

File: pkg/meta/dynamodb/dynamodb.go [hunk 2]
```diff
@@ -86,6 +93,10 @@ func New(client *dynamodb.Client, params DBDriverParameters, log log.Logger) (*D
 	return &dynamoWrapper, nil
 }
 
+func (dwr *DynamoDB) SignatureStorage() mTypes.SignatureStorage {
+	return dwr.SigStore
```
Comment (peusebiu) [inline pkg/meta/dynamodb/dynamodb.go:97]: you can make it private if there is an exported method to retrieve it.

File: pkg/meta/dynamodb/dynamodb.go [hunk 3]
```diff
@@ -658,7 +669,7 @@ func (dwr *DynamoDB) UpdateSignaturesValidity(repo string, manifestDigest godige
 			layersInfo := []mTypes.LayerInfo{}
 
 			for _, layerInfo := range sigInfo.LayersInfo {
-				author, date, isTrusted, _ := signatures.VerifySignature(sigType, layerInfo.LayerContent, layerInfo.SignatureKey,
+				author, date, isTrusted, _ := dwr.SigStore.VerifySignature(sigType, layerInfo.LayerContent, layerInfo.SignatureKey,
```
Comment (peusebiu) [inline pkg/meta/dynamodb/dynamodb.go:672]: use the method

File: pkg/meta/dynamodb/dynamodb.go (hunk 2 above)
Comment (peusebiu) [reply pkg/meta/dynamodb/dynamodb.go:97]: hmm, do we use this method anywhere?

File: pkg/api/routes.go [hunk 4]
```diff
@@ -193,7 +193,7 @@ func (rh *RouteHandler) SetupRoutes() {
 	// Preconditions for enabling the actual extension routes are part of extensions themselves
 	ext.SetupSearchRoutes(rh.c.Config, prefixedRouter, rh.c.StoreController, rh.c.MetaDB, rh.c.CveInfo,
 		rh.c.Log)
-	ext.SetupImageTrustRoutes(rh.c.Config, prefixedRouter, rh.c.Log)
+	ext.SetupImageTrustRoutes(rh.c.Config, rh.c.MetaDB, prefixedRouter, rh.c.Log)
```
Comment (andaaron) [inline pkg/api/routes.go:196]: Nitpicking, maybe the router should be the 2nd parameter and the db the 3rd, similar to the other methods.

File: pkg/meta/dynamodb/dynamodb_test.go
Comment (andaaron) [inline pkg/meta/dynamodb/dynamodb_test.go:1]: Does this file need a build label for a specific extension?

**Review by andaaron — COMMENTED** [review]
Overall this is very good PR.

**Comment by peusebiu** [comment]
lgtm

**Review by andaaron — APPROVED** [review]

**Review by rchincha — APPROVED** [review]
lgtm

**Comment by rchincha** [comment]
@Andreea-Lupu  pls rebase?

**Comment by andaaron** [comment]
Superseded by https://github.com/project-zot/zot/pull/1744/files

**Pull request closed without merging**